pgjdbc/src/test/java/org/postgresql/test/util/LruCacheTest.java

```diff
@@ -104,7 +105,9 @@ public void testEvictsLeastRecentlyUsed() throws SQLException {
     a = use(1);
     b = use(2);
     c = use(3);
-    a = use(1); // reuse a
+    //calling use actually inserts value into map, which tests eviction based
+    //on insertion order rather than USED order.
+    assertSame(a, cache.get(1));
```

Comment:
The test was right modulo the definition of `use`.

Please refrain from modifying the existing tests.

I agree the naming was not the best here, however, `LruCache + use` did the right thing here.
LruCache was coded for `borrow+return` pattern, and `use` method here follows exactly that. That is why the cache is LRU for this definition of `use`.

Reply:
I restored the original test and kept this version with names highlighting the different access patterns.

Reply:
Am I right the very same test passes even without changes to `LruCache`?